hack/jenkins/e2e.sh

```diff
@@ -21,195 +21,96 @@ set -o errexit
-  KUBERNETES_PROVIDER="gke"
-  : ${E2E_ZONE:="us-central1-f"}
+  export KUBERNETES_PROVIDER="gke"
+  export ZONE="us-central1-f"
   # By default, GKE tests run against the GKE test endpoint using CI Cloud SDK.
   # Release jobs (e.g. prod, staging, and test) override these two variables.
-  : ${CLOUDSDK_BUCKET:="gs://cloud-sdk-build/testing/staging"}
-  : ${GKE_API_ENDPOINT:="https://test-container.sandbox.googleapis.com/"}
+  export CLOUDSDK_BUCKET="gs://cloud-sdk-build/testing/staging"
+  export CLOUDSDK_API_ENDPOINT_OVERRIDES_CONTAINER="https://test-container.sandbox.googleapis.com/"
+  export FAIL_ON_GCP_RESOURCE_LEAK="true"
 elif [[ ${JOB_NAME} =~ ^kubernetes-.*-aws ]]; then
-  KUBERNETES_PROVIDER="aws"
-  : ${E2E_MIN_STARTUP_PODS:="1"}
-  : ${E2E_ZONE:="us-east-1a"}
-  : ${NUM_NODES_PARALLEL:="6"}  # Number of nodes required to run all of the tests in parallel
+  export KUBERNETES_PROVIDER="aws"
+  export E2E_MIN_STARTUP_PODS="1"
+  export KUBE_AWS_ZONE="us-west-2a"
+  export MASTER_SIZE="m3.medium"
+  export NODE_SIZE="m3.medium"
+  export NUM_NODES="3"
 fi
 
-if [[ "${KUBERNETES_PROVIDER}" == "aws" ]]; then
-  if [[ "${PERFORMANCE:-}" == "true" ]]; then
-    : ${MASTER_SIZE:="m3.xlarge"}
-    : ${NUM_NODES:="100"}
-    : ${GINKGO_TEST_ARGS:="--ginkgo.focus=\[Feature:Performance\]"}
-  else
-    : ${MASTER_SIZE:="m3.medium"}
-    : ${NODE_SIZE:="m3.medium"}
-    : ${NUM_NODES:="3"}
-  fi
+# Set environment variables based on soak jobs
+if [[ ${JOB_NAME} =~ soak-weekly ]]; then
+  export FAIL_ON_GCP_RESOURCE_LEAK="false"
+  export E2E_TEST="false"
+  export E2E_DOWN="false"
+elif [[ ${JOB_NAME} =~ soak-continuous ]]; then
+  export FAIL_ON_GCP_RESOURCE_LEAK="false"
+  export E2E_UP="false"
+  export E2E_DOWN="false"
+  export E2E_CLEAN_START="true"
+  # We should be testing the reliability of a long-running cluster. The
+  # [Disruptive] tests kill/restart components or nodes in the cluster,
+  # defeating the purpose of a soak cluster. (#15722)
+  export GINKGO_TEST_ARGS="--ginkgo.skip=\[Disruptive\]|\[Flaky\]|\[Feature:.+\]"
 fi
 
-# CURRENT_RELEASE_PUBLISHED_VERSION is the JENKINS_PUBLISHED_VERSION for the
-# release we are currently pointing our release testing infrastructure at.
-# When 1.2.0-beta.0 comes out, e.g., this will become "ci/latest-1.2"
-CURRENT_RELEASE_PUBLISHED_VERSION="ci/latest-1.1"
+# Set environment variables based on upgrade jobs
+if [[ ${JOB_NAME} =~ ^kubernetes-upgrade ]]; then
+  # Upgrade jobs bounce back and forth between versions; just force
+  # it to always get the tars of the version it wants to test.
+  export JENKINS_FORCE_GET_TARS="y"
+  export FAIL_ON_GCP_RESOURCE_LEAK="false"
+  if [[ "${KUBERNETES_PROVIDER}" == "gce" ]]; then
+    export NUM_NODES=5
+  fi
+  if [[ "${JOB_NAME}" =~ step1 ]]; then
+    export E2E_TEST="false"
+    export E2E_DOWN="false"
+  elif [[ "${JOB_NAME}" =~ step2 ]]; then
+    export E2E_OPT="--check_version_skew=false"
+    export E2E_UP="false"
```

Comment:
minor suggestion: maybe in the outer if block ("if kubernetes-upgrade") set `E2E_UP="false"` and `E2E_DOWN="false"` - then you only need to override for step1 and step7.